iceoryx2-ffi/ffi/src/api/config.rs

```diff
@@ -12,22 +12,1639 @@
+#[no_mangle]
+pub unsafe extern "C" fn iox2_config_defaults_publish_subscribe_publisher_history_size(
+    handle: iox2_config_ref_h,
+) -> c_size_t {
+    debug_assert!(!handle.is_null());
+
+    let config = &*handle.as_type();
+    config
+        .value
+        .as_ref()
+        .value
+        .defaults
+        .publish_subscribe
+        .publisher_history_size
+}
+
+/// Sets the maximum history size a [`iox2_subscriber_h`](crate::api::iox2_subscriber_h) can
+/// request from a [`iox2_publisher_h`](crate::api::iox2_publisher_h).
+///
+/// # Safety
+///
+/// * `handle` - A valid non-owning [`iox2_config_ref_h`] obtained by [`iox2_cast_config_ref_h`].
+#[no_mangle]
+pub unsafe extern "C" fn iox2_config_defaults_publish_subscribe_set_publisher_history_size(
+    handle: iox2_config_ref_h,
+    value: c_size_t,
+) {
+    debug_assert!(!handle.is_null());
+
+    let config = &mut *handle.as_type();
+    config
+        .value
+        .as_mut()
+        .value
+        .defaults
+        .publish_subscribe
+        .publisher_history_size = value;
+}
+
+/// Defines how the [`iox2_subscriber_h`](crate::api::iox2_subscriber_h) buffer behaves when it is
+/// full. When safe overflow is activated, the [`iox2_publisher_h`](crate::api::iox2_publisher_h) will
+/// replace the oldest [`iox2_sample_h`](crate::api::iox2_sample_h) with the newest one.
+///
+/// # Safety
+///
+/// * `handle` - A valid non-owning [`iox2_config_ref_h`] obtained by [`iox2_cast_config_ref_h`].
+#[no_mangle]
+pub unsafe extern "C" fn iox2_config_defaults_publish_subscribe_enable_safe_overflow(
+    handle: iox2_config_ref_h,
+) -> bool {
+    debug_assert!(!handle.is_null());
+
+    let config = &*handle.as_type();
+    config
+        .value
+        .as_ref()
+        .value
+        .defaults
+        .publish_subscribe
+        .enable_safe_overflow
+}
+
+/// Enables/disables safe overflow
+///
+/// # Safety
+///
+/// * `handle` - A valid non-owning [`iox2_config_ref_h`] obtained by [`iox2_cast_config_ref_h`].
+#[no_mangle]
+pub unsafe extern "C" fn iox2_config_defaults_publish_subscribe_set_enable_safe_overflow(
+    handle: iox2_config_ref_h,
+    value: bool,
+) {
+    debug_assert!(!handle.is_null());
+
+    let config = &mut *handle.as_type();
+    config
+        .value
+        .as_mut()
+        .value
+        .defaults
+        .publish_subscribe
+        .enable_safe_overflow = value;
+}
+
+/// If safe overflow is deactivated it defines the deliver strategy of the
+/// [`iox2_publisher_h`](crate::api::iox2_publisher_h) when the
+/// [`iox2_subscriber_h`](crate::api::iox2_subscriber_h)s buffer is full.
+///
+/// Returns [`iox2_unable_to_deliver_strategy_e`]
+///
+/// # Safety
+///
+/// * `handle` - A valid non-owning [`iox2_config_ref_h`] obtained by [`iox2_cast_config_ref_h`].
+#[no_mangle]
+pub unsafe extern "C" fn iox2_config_defaults_publish_subscribe_unable_to_deliver_strategy(
+    handle: iox2_config_ref_h,
+) -> c_int {
```

Comment:
Would it make sense to return the iox2_unable_to_deliver_strategy_e enum directly?

Reply:
Maybe, you never did this in any other API so I just followed the other code for consistency reasons.

Reply:
Ah, right. It's an error enum and then one cannot return `OK` anymore